### PR TITLE
[FIX] type printing for references

### PR DESCRIPTION
**Pull request by h-2**
It was broken for const-ness and references which both weren't tested...

### CODE REVIEW

File: include/seqan3/core/detail/type_inspection.hpp [hunk 1]
```diff
@@ -41,18 +41,28 @@ namespace seqan3::detail
  * \note The returned name is implementation defined and might change between different tool chains.
  */
 template <typename type>
-inline static const std::string type_name_as_string = [] ()
+inline std::string const type_name_as_string = [] ()
 {
+    std::string ret;
```
Comment (rrahn) [inline include/seqan3/core/detail/type_inspection.hpp]:
```suggestion
    std::string demangled_name{};
```

**Comment by rrahn** [comment]
[L33](https://github.com/seqan/seqan3/pull/1450/files#diff-89986d8c742bee39cd1e3524ecbf3cd5L33) uses the wrong doxygen command. This should be `\details` I am wondering why this wasn't detected before 🤔

File: include/seqan3/core/detail/type_inspection.hpp (hunk 1 above)
Comment (h-2) [reply include/seqan3/core/detail/type_inspection.hpp]: I used `ret` throughout the library for the return value of lambda expressions, especially in the context of immediately evaluated lambdas used in the initialisation of variables.
But I can change it here, it's not important.

**Comment by h-2** [comment]
> L35 uses the wrong doxygen command. This should be \details I am wondering why this wasn't detected before thinking

Fixed it while I was there.

File: include/seqan3/core/detail/type_inspection.hpp [hunk 2]
```diff
@@ -41,18 +41,28 @@ namespace seqan3::detail
  * \note The returned name is implementation defined and might change between different tool chains.
  */
 template <typename type>
-inline static const std::string type_name_as_string = [] ()
+inline std::string const type_name_as_string = [] ()
 {
+    std::string demangled_name;
```
Comment (rrahn) [inline include/seqan3/core/detail/type_inspection.hpp]:
```suggestion
    std::string demangled_name{};
```

Comment (h-2) [reply include/seqan3/core/detail/type_inspection.hpp]: Our styleguide says:

> initialise all variables upon declaration, unless you really know what you are doing

I know what I am doing :laughing:

**Comment by h-2** [comment]
Please squash this then.

**Pull request merged**